org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -705,14 +807,46 @@ public void testTryFinallyMergeSameBranch() {
 	}
 
 	@Test
-	public void testTryFinallyMergeDifferentBranches() {
+	public void try_finally_with_filter_should_show_full_branch_coverage_when_different_branches_are_covered_and_merged() {
 		createTryFinally();
 		probes[0] = true;
 		probes[3] = true;
 		runMethodAnalzer(TRY_FINALLY_FILTER);
 		assertLine(1001, 2, 3, 0, 2);
 	}
 
+	// === Scenario: descending line numbers ===
+
+	private void createDescendingLineNumbers() {
+		final Label l0 = new Label();
+		method.visitLabel(l0);
+		method.visitLineNumber(1003, l0);
+		method.visitInsn(Opcodes.NOP);
+		method.visitInsn(Opcodes.NOP);
+		method.visitInsn(Opcodes.NOP);
+		final Label l1 = new Label();
+		method.visitLabel(l1);
+		method.visitLineNumber(1002, l1);
+		method.visitInsn(Opcodes.NOP);
+		method.visitInsn(Opcodes.NOP);
+		final Label l2 = new Label();
+		method.visitLabel(l2);
+		method.visitLineNumber(1001, l2);
+		method.visitInsn(Opcodes.RETURN);
+	}
+
+	@Test
+	public void decending_line_numbers_should_report_lines_correctly() {
```

Comment:
Out of curiosity I'm wondering what motivated you to add this test?

Reply:
@Godin To get full branch coverage on `MethodAnalyzer.visitLineNumber()`. Any proposal for a different testing approach or improved test name?

Reply:
@marchof fine like it is, I was just curious.